src/FrequencyMapAgent.cpp

```diff
@@ -248,17 +262,27 @@ namespace geopm
         m_do_write_batch = false;
 
         if (!m_is_adjust_initialized) {
+            std::cout << "m_is_adjust_initialized = false" << std::endl;
             // adjust all controls once in case not applied by policy
             for (size_t ctl_idx = 0; ctl_idx < (size_t) m_num_freq_ctl_domain; ++ctl_idx) {
                 double val = m_platform_io.read_signal("CPU_FREQUENCY_MAX_CONTROL", m_freq_ctl_domain_type, ctl_idx);
                 m_platform_io.adjust(m_freq_control_idx[ctl_idx], val);
             }
             m_platform_io.adjust(m_uncore_min_ctl_idx, m_uncore_init_min);
             m_platform_io.adjust(m_uncore_max_ctl_idx, m_uncore_init_max);
+
+            std::cout << "number of gpu ctl domains: " << m_num_gpu_freq_ctl_domain << std::endl;
+            for(size_t ctl_idx = 0; ctl_idx < (size_t)m_num_gpu_freq_ctl_domain; ++ctl_idx) {
```

Comment:
I would advise against using C-style casts. However, I believe you should be able to get away with making ``ctl_idx`` ``int`` to avoid having to do it. If you do have to perform a cast, I recommend using C++'s cast operators (e.g. ``static_cast<>``).

Update: actually, I'm seeing this same cast is sprinkled all over this file. At this point just leave it consistent with the rest of the code for now. I think it's out of the scope of this PR to address this.

Nit: ``for (`` (space after ``for`` keyword)